### PR TITLE
[Tests]Remove usage of WithRunning as it is deprecated

### PR DESCRIPTION
**Pull request by Barakmor1**
Use WithRunStrategy instead to create a running VM in the recommended way.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/libvmi/vm.go
Comment (akalenyu) [inline pkg/libvmi/vm.go:66]: I guess we can remove the linter rule? https://github.com/kubevirt/kubevirt/commit/752a550cb501b61fa832a03c7b454fc68c1c6f0e#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9

File: tests/monitoring/metrics.go
Comment (xpivarc) [inline tests/monitoring/metrics.go:145]: Not related to your PR but we should replace this custom function with one that is widely used.

Comment (Barakmor1) [reply tests/monitoring/metrics.go:145]: Yes, that makes sense to me.

File: pkg/libvmi/vm.go
Comment (Barakmor1) [reply pkg/libvmi/vm.go:66]: Thanks for the note.
done

**Comment by Barakmor1** [comment]
/hold 
waiting for https://github.com/kubevirt/kubevirt/pull/12895 to get in

**Review by jean-edouard — COMMENTED** [review]
/approve

File: .golangci.yml
Comment (fossedihelm) [inline .golangci.yml:126]: I think you dropped too many lines

**Review by fossedihelm — CHANGES_REQUESTED** [review]

File: .golangci.yml
Comment (Barakmor1) [reply .golangci.yml]: Done :)

**Comment by fossedihelm** [comment]
Just waiting for the related PR to gets in, otherwise LGTM

**Pull request merged**
